aten/src/ATen/core/enum_type.h

```diff
@@ -49,6 +49,7 @@ struct TORCH_API EnumType : public NamedType {
 
   bool equals(const Type& rhs) const override {
     if (auto* enum_rhs = rhs.castRaw<EnumType>()) {
+      // NOLINTNEXTLINE(bugprone-unchecked-optional-access)
       return name().value() == enum_rhs->name().value() &&
```

Comment:
```suggestion
      return name() == enum_rhs->name() &&
```
Is there a default value specified to either optional otherwise? If so, it should be equivalent. If not, should be UB?

Reply:
We can't use this optimisation because the old code would throw, which may be intentional?